pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -771,6 +771,12 @@ type CommonPrometheusFields struct {
 	// +listType=map
 	// +listMapKey=name
 	ScrapeClasses []ScrapeClass `json:"scrapeClasses,omitempty"`
+
+	// Defines the service discovery role used to discover targets from `ServiceMonitor` objects.
+	// If set, the value should be either "Endpoints" or "EndpointSlice".
+	// If unset, the operator assumes the "Endpoints" role.
+	// +optional
+	ServiceDiscoveryRole *string `json:"serviceDiscoveryRole,omitempty"`
```

Comment:
```suggestion
	ServiceDiscoveryRole *ServiceDiscoveryRole `json:"serviceDiscoveryRole,omitempty"`
```

Let's use the newly created Enum :)